### PR TITLE
First aid proficiencies

### PR DESCRIPTION
**Pull request by nexusmrsep**
#### Summary
Content "First aid proficiencies"

#### Purpose of change
Introducing first aid proficiencies & low tier practice recipes for healthcare.

#### Describe the solution
Two proficiencies:
- First Aid - "You know how to bandage wounds and understand basic principles of wound care."
- First Aid Expert  - "Your extensive field experience in bandaging and wound care is on par with that of a paramedic."

Effects: 
- first-tier is equal to extra 1 level of health care (firstaid) skill in efficiency calculations for bandaging & disinfecting & applying pressure to wounds & examining NPC's wounds
- second-tier is equal to 2 levels of health care (firstaid) skill in efficiency calculations for bandaging & disinfecting & applying pressure to wounds & examining NPC's wounds
- both bonuses cumulate for up to +3 extra lvl bonuses
- each tier upgrades tourniquet efficiency from base 2/3 to 3/4 and 4/5 efficiency of blood loss reduction
- each tier reduces application time of bandages and disinfectants (cleaning wounds) by half (up to 1/4 of base) (*AFTER AUDIT)

* - application times were audited, so the ridiculously low time required to apply was significantly raised to more sane values

Two recipes:
- bandaging (beginner) - "Practice bandaging imaginary wounds while making sure your bandage won't fall off during activities."
- bandaging (advanced) - "Practice bandaging different limbs and less accessible parts of your body using both hands and each of your hands alone."

Both are low tier practice recipes, because in general bandaging has quite limited impact on general health care skill.

New proficiencies were added where applicable to professions and hobbies - especially medical and paramedical ones.

#### Describe alternatives you've considered
Adding some higher tier recipes using some less then ethical requirements.
@I-am-Erk  can you shine some light at the topic and perhaps tell what could be the next step from here? Any ideas for expanding from this, like for example destructive use of 'higher' tier items as saline solution, disinfectant, splints, tourniquets, scalpels and maybe some other fancy lab tools like microscopes etc. to introduce mid to high tier practice recipes?

#### Testing
Light testing in game. Compiled. Proficiencies visible, applied in hobbies and professions. Trainable via practice recipes. Boost in first aid items quality visible on both tiers, time of application of first aid items reduced on both tiers.

#### Additional context
Inspired by a [Reddit discussion](https://www.reddit.com/r/cataclysmdda/comments/qztwkl/how_do_you_train_health_care_and_other_skills_in/).

### CODE REVIEW

**Comment by Termineitor244** [comment]
Perfect! Now we only need mechanics and applied science practice recipes to eliminate the most difficult skill grinds of the game! (Food and Survival still don't have practice recipes, but they are way easier to level up and/or make practice recipes).

Just a little suggestion, but for training to level 3 I would expect to have to train first aid skills in general, not just bandaging, the suggestion of Vormithrax on the reddit thread is good in that sense, just with the problem that it needs a body... Why not train those things in your own body?

**Comment by Venera3** [comment]
I'd add bandaging as a two-tier proficiency and route the practice through that. Healthcare as a skill governs too disparate things (from NPC diagnostics to making bootleg medication) that bandaging your way up to professional level would feel janky - I am aware that that's how it works *now*, but it really shouldn't. Not that practicing on make-believe wounds should get you far, but it's probably worth it to close the incentive to self-mutilate ans bandage.

**Comment by nexusmrsep** [comment]
I've updated this PR by adding first aid proficiencies, as suggested. I still need to do some cleanup and testing to ensure quality.

**Comment by ISuckM8** [comment]
How many bandages do you have to use to get the proficiencies?

**Comment by VoleurVolant** [comment]
I would strongly suggest adding training for the first level of first aid proficiency to the Paramedics (mag_firstaid) entry and the second level to the Guide to Advanced Emergency Care (emergency_book) entry in data/json/items/book/firstaid.json. Practice recipes are helpful, and I wouldn't necessarily remove them, but having meaningful resources to learn from makes far more sense to me.

**Comment by I-am-Erk** [comment]
So, I am not super keen on calling this "first aid". That's too general. I would suggest we call these "wound care" and "advanced wound care" perhaps, so they might later affect disinfection and stuff but not, for example, airway management. The term first aid makes it hard to tell the difference between the proficiency and the skill, they cover all the same stuff.

Otherwise, I think what you've got here is good, without looking deeply at the code itself. A later proficiency could be related to bone setting and splinting. I don't recall many other things we presently use healthcare for in game.

**Comment by nexusmrsep** [comment]
> How many bandages do you have to use to get the proficiencies?

Assuming that you can reuse the rag(s) that you get as a byproduct to make makeshift bandage(s) you can even manage with one, but it would be convenient to have at least a few to chain training 'sessions' if you so wish.

> (...) having meaningful resources to learn from makes far more sense to me.

Good catch @VoleurVolant , I'm adding it.

> So, I am not super keen on calling this "first aid". 

@I-am-Erk  - I understand where are you coming from, and in real world 'first aid' term would be indeed wide and would cover many procedures. 

But as far as I can perceive, there will never be any airway management, CPR, heart massage, et cetera in general "first aid" field available for the player. The options are very limited, mainly because in 95% of cases it needs to be self-applied. Perhaps in the future such procedures would be implemented for the benefit of saving NPC's but I don't believe so. Thus one can be a MD or a Paramedic, but I can't see field tracheotomy and similar more 'invasive' or rather "from-second-party" first aid procedures to be applicable by the character on the character. In conclusion: I think it is safe to use a broader term here.

However I don't want the dispute over the use of a term to block this feature so I'm changing it on your request.

I can imagine that in the future (hopefully Soon™) when the wound system would track every wound and wound care would become more in-depth, so the general 'health care' skill can be expanded with proficiencies like 'surgery' (autodoc for the win, but also 'self surgery' like stitching wounds, removing foreign bodies from wounds etc.), perhaps even some aspects of  'pharmacology' etc.

**Comment by kevingranade** [comment]
I absolutely aspire to making first aid and dragging allies back to base for further treatment a very real thing that happens in the game, including having the player character downed and similarly rescued.

This is the kind of game loop that we can dig into if NPC allies are a first-order game feature and the player can switch control to a NPC while waiting for the original character to recover from their debilitating injury or injuries.

File: data/json/items/book/firstaid.json [hunk 1]
```diff
@@ -16,6 +16,7 @@
     "required_level": 5,
     "max_level": 8,
     "intelligence": 10,
+    "proficiencies": [ { "proficiency": "prof_firstaid_expert", "time_factor": 0.75, "fail_factor": 0.5 } ],
```
Comment (kevingranade) [inline data/json/items/book/firstaid.json:19]:
```suggestion
    "proficiencies": [ { "proficiency": "prof_wound_care_expert", "time_factor": 0.75, "fail_factor": 0.5 } ],
```

File: data/json/items/book/firstaid.json [hunk 2]
```diff
@@ -35,6 +36,7 @@
     "skill": "firstaid",
     "max_level": 1,
     "intelligence": 6,
+    "proficiencies": [ { "proficiency": "prof_firstaid", "time_factor": 0.75, "fail_factor": 0.5 } ],
```
Comment (kevingranade) [inline data/json/items/book/firstaid.json:39]:
```suggestion
    "proficiencies": [ { "proficiency": "prof_wound_care", "time_factor": 0.75, "fail_factor": 0.5 } ],
```

File: data/json/items/book/firstaid.json [hunk 3]
```diff
@@ -54,6 +56,10 @@
     "skill": "firstaid",
     "max_level": 3,
     "intelligence": 8,
+    "proficiencies": [
+      { "proficiency": "prof_firstaid", "time_factor": 0.75, "fail_factor": 0.5 },
+      { "proficiency": "prof_firstaid_expert", "time_factor": 0.75, "fail_factor": 0.5 }
```
Comment (kevingranade) [inline data/json/items/book/firstaid.json:61]:
```suggestion
      { "proficiency": "prof_wound_care", "time_factor": 0.75, "fail_factor": 0.5 },
      { "proficiency": "prof_wound_care_expert", "time_factor": 0.75, "fail_factor": 0.5 }
```

File: data/json/items/book/firstaid.json [hunk 4]
```diff
@@ -71,6 +77,7 @@
     "skill": "firstaid",
     "max_level": 2,
     "intelligence": 6,
+    "proficiencies": [ { "proficiency": "prof_firstaid", "time_factor": 0.75, "fail_factor": 0.5 } ],
```
Comment (kevingranade) [inline data/json/items/book/firstaid.json:80]:
```suggestion
    "proficiencies": [ { "proficiency": "prof_wound_care", "time_factor": 0.75, "fail_factor": 0.5 } ],
```

File: data/json/professions.json [hunk 5]
```diff
@@ -1149,6 +1149,7 @@
     "description": "Fresh out of med school, you've got little in the way of practical experience and just a handful of first-aid supplies.  You just hope it will be enough if 'physician, heal thyself' turns out to be more literal than you expected.",
     "points": 2,
     "skills": [ { "level": 4, "name": "firstaid" } ],
+    "proficiencies": [ "prof_firstaid", "prof_firstaid_expert" ],
```
Comment (kevingranade) [inline data/json/professions.json:1152]:
```suggestion
    "proficiencies": [ "prof_wound_care", "prof_wound_care_expert" ],
```

File: data/json/professions.json [hunk 6]
```diff
@@ -1258,6 +1259,7 @@
     "description": "You went on providing in-home care for the elderly even as the whole world fell apart around you.  You can only pray that you don't see your former clients among the walking dead…",
     "points": 1,
     "skills": [ { "level": 2, "name": "firstaid" }, { "level": 1, "name": "cooking" } ],
+    "proficiencies": [ "prof_firstaid" ],
```
Comment (kevingranade) [inline data/json/professions.json:1262]:
```suggestion
    "proficiencies": [ "prof_wound_care" ],
```

File: data/json/professions.json [hunk 7]
```diff
@@ -1839,6 +1841,7 @@
     "description": "On your way to respond to an emergency call, you nearly drove straight into a riot in the city.  Turning off of the burning, debris-covered streets, you took a long detour only to find yourself lost.  That call will have to wait - you're in an emergency of your own now.",
     "points": 5,
     "skills": [ { "level": 3, "name": "firstaid" }, { "level": 3, "name": "driving" }, { "level": 1, "name": "electronics" } ],
+    "proficiencies": [ "prof_firstaid" ],
```
Comment (kevingranade) [inline data/json/professions.json:1844]:
```suggestion
    "proficiencies": [ "prof_wound_care" ],
```

File: data/json/professions.json [hunk 8]
```diff
@@ -3795,6 +3798,7 @@
       { "level": 1, "name": "electronics" },
       { "level": 3, "name": "firstaid" }
     ],
+    "proficiencies": [ "prof_firstaid", "prof_firstaid_expert" ],
```
Comment (kevingranade) [inline data/json/professions.json:3801]:
```suggestion
    "proficiencies": [ "prof_wound_care", "prof_wound_care_expert" ],
```

File: data/json/professions.json [hunk 9]
```diff
@@ -3813,6 +3817,7 @@
     "description": "You were separated from your partner while out on a call.  You managed to hang onto some medical supplies, but it's looking like the only life that needs saving now is yours.",
     "points": 3,
     "skills": [ { "level": 2, "name": "driving" }, { "level": 1, "name": "mechanics" }, { "level": 5, "name": "firstaid" } ],
+    "proficiencies": [ "prof_firstaid", "prof_firstaid_expert" ],
```
Comment (kevingranade) [inline data/json/professions.json:3820]:
```suggestion
    "proficiencies": [ "prof_wound_care", "prof_wound_care_expert" ],
```

File: data/json/professions.json [hunk 10]
```diff
@@ -3852,7 +3857,7 @@
       { "level": 4, "name": "firstaid" }
     ],
     "traits": [ "PROF_MED" ],
-    "proficiencies": [ "prof_spotting" ],
+    "proficiencies": [ "prof_spotting", "prof_firstaid", "prof_firstaid_expert" ],
```
Comment (kevingranade) [inline data/json/professions.json:3860]:
```suggestion
    "proficiencies": [ "prof_spotting", "prof_wound_care", "prof_wound_care_expert" ],
```

File: data/json/professions.json [hunk 11]
```diff
@@ -4005,6 +4010,7 @@
     "description": "You were deployed to autopsy one of the rioters showing feral behavior before being put down.  When they got back up, you knew this was out of your job description.",
     "points": 5,
     "skills": [ { "level": 5, "name": "firstaid" } ],
+    "proficiencies": [ "prof_firstaid", "prof_firstaid_expert" ],
```
Comment (kevingranade) [inline data/json/professions.json:4013]:
```suggestion
    "proficiencies": [ "prof_wound_care", "prof_wound_care_expert" ],
```

File: data/json/proficiencies/health_care.json [hunk 12]
```diff
@@ -0,0 +1,23 @@
+[
+  {
+    "type": "proficiency",
+    "id": "prof_firstaid",
```
Comment (kevingranade) [inline data/json/proficiencies/health_care.json:4]:
```suggestion
    "id": "prof_wound_care",
```

File: data/json/recipes/practice/health.json [hunk 13]
```diff
@@ -0,0 +1,51 @@
...
+    "type": "practice",
+    "activity_level": "NO_EXERCISE",
+    "category": "CC_PRACTICE",
+    "subcategory": "CSC_PRACTICE_HEALTH",
+    "name": "bandaging (beginner)",
+    "description": "Practice bandaging imaginary wounds while making sure your bandage won't fall off during activities.",
+    "skill_used": "firstaid",
+    "time": "1 h",
+    "practice_data": { "min_difficulty": 0, "max_difficulty": 1, "skill_limit": 1 },
+    "proficiencies": [ { "proficiency": "prof_firstaid", "fail_multiplier": 1, "time_multiplier": 1 } ],
```
Comment (kevingranade) [inline data/json/recipes/practice/health.json:13]:
```suggestion
    "proficiencies": [ { "proficiency": "prof_wound_care", "fail_multiplier": 1, "time_multiplier": 1 } ],
```

File: data/json/recipes/practice/health.json [hunk 14]
```diff
@@ -0,0 +1,51 @@
...
+    "type": "practice",
+    "activity_level": "NO_EXERCISE",
+    "category": "CC_PRACTICE",
+    "subcategory": "CSC_PRACTICE_HEALTH",
+    "name": "bandaging (advanced)",
+    "description": "Practice bandaging different limbs and less accessible parts of your body using both hands and each of your hands alone.",
+    "skill_used": "firstaid",
+    "time": "1 h",
+    "practice_data": { "min_difficulty": 1, "max_difficulty": 2, "skill_limit": 3 },
+    "proficiencies": [ { "proficiency": "prof_firstaid_expert", "fail_multiplier": 1, "time_multiplier": 1 } ],
```
Comment (kevingranade) [inline data/json/recipes/practice/health.json:37]:
```suggestion
    "proficiencies": [ { "proficiency": "prof_wound_care_expert", "fail_multiplier": 1, "time_multiplier": 1 } ],
```

File: data/mods/package_bionic_professions/bionic_professions.json [hunk 15]
```diff
@@ -243,7 +243,7 @@
     "description": "When bionics first emerged, you were quick to make them into your career, and spent your days overseeing their installation.  That makes you one of the few non-zombies in the world that can calibrate an Autodoc, which might come in handy.",
     "points": 4,
     "skills": [ { "level": 4, "name": "firstaid" }, { "level": 4, "name": "electronics" } ],
-    "proficiencies": [ "prof_intro_biology", "prof_physiology" ],
+    "proficiencies": [ "prof_intro_biology", "prof_physiology", "prof_firstaid" ],
```
Comment (kevingranade) [inline data/mods/package_bionic_professions/bionic_professions.json:246]:
```suggestion
    "proficiencies": [ "prof_intro_biology", "prof_physiology", "prof_wound_care" ],
```

File: data/mods/package_bionic_professions/bionic_professions.json [hunk 16]
```diff
@@ -274,7 +274,7 @@
     "points": 6,
     "CBMs": [ "bio_surgical_razor", "bio_flashlight", "bio_batteries", "bio_power_storage_mkII" ],
     "skills": [ { "level": 8, "name": "firstaid" } ],
-    "proficiencies": [ "prof_intro_biology", "prof_physiology" ],
+    "proficiencies": [ "prof_intro_biology", "prof_physiology", "prof_firstaid", "prof_firstaid_expert" ],
```
Comment (kevingranade) [inline data/mods/package_bionic_professions/bionic_professions.json:277]:
```suggestion
    "proficiencies": [ "prof_intro_biology", "prof_physiology", "prof_wound_care", "prof_wound_care_expert" ],
```

File: doc/PROFICIENCY_LIST.md [hunk 17]
```diff
@@ -124,6 +124,9 @@
 | prof_spotting | Spotting and Awareness | You are skilled at spotting things out of the ordinary, like traps or ambushes. | None
 | |
 | prof_parkour | Parkour Expert | You're skilled at clearing obstacles; terrain like railings or counters are as easy for you to move on as solid ground. | None
+| |
+| prof_firstaid | First Aid | You know how to bandage wounds and understand basic principles of wound care. | None
+| prof_firstaid_expert | First Aid Expert | Your extensive field experience in bandaging and wound care is on par with that of a paramedic. | prof_firstaid
```
Comment (kevingranade) [inline doc/PROFICIENCY_LIST.md:129]:
```suggestion
| prof_wound_care | Wound Care | You know how to bandage wounds and understand basic principles of wound care. | None
| prof_wound_care_expert | Wound Care Expert | Your extensive field experience in bandaging and wound care is on par with that of a paramedic. | prof_firstaid
```

File: src/character.cpp [hunk 18]
```diff
@@ -336,6 +336,8 @@ static const move_mode_id move_mode_walk( "walk" );
 
 static const mtype_id mon_player_blob( "mon_player_blob" );
 
+static const proficiency_id proficiency_prof_firstaid( "prof_firstaid" );
+static const proficiency_id proficiency_prof_firstaid_expert( "prof_firstaid_expert" );
```
Comment (kevingranade) [inline src/character.cpp:340]:
```suggestion
static const proficiency_id proficiency_prof_wound_care( "prof_wound_care" );
static const proficiency_id proficiency_prof_wound_care_expert( "prof_wound_care_expert" );
```

File: src/game.cpp [hunk 19]
```diff
@@ -251,6 +251,8 @@ static const mtype_id mon_manhack( "mon_manhack" );
 
 static const overmap_special_id overmap_special_world( "world" );
 
+static const proficiency_id proficiency_prof_firstaid( "prof_firstaid" );
+static const proficiency_id proficiency_prof_firstaid_expert( "prof_firstaid_expert" );
 static const proficiency_id proficiency_prof_parkour( "prof_parkour" );
```
Comment (kevingranade) [inline src/game.cpp:256]:
```suggestion
static const proficiency_id proficiency_prof_parkour( "prof_parkour" );
static const proficiency_id proficiency_prof_wound_care( "prof_wound_care" );
static const proficiency_id proficiency_prof_wound_care_expert( "prof_wound_care_expert" );

```

File: src/game.cpp [hunk 20]
```diff
@@ -5096,9 +5098,11 @@ bool game::npc_menu( npc &who )
         }
     } else if( choice == examine_wounds ) {
         ///\EFFECT_PER slightly increases precision when examining NPCs' wounds
-
         ///\EFFECT_FIRSTAID increases precision when examining NPCs' wounds
-        const bool precise = u.get_skill_level( skill_firstaid ) * 4 + u.per_cur >= 20;
+        int prof_bonus = u.get_skill_level( skill_firstaid );
+        prof_bonus = u.has_proficiency( proficiency_prof_firstaid ) ? prof_bonus + 1 : prof_bonus;
+        prof_bonus = u.has_proficiency( proficiency_prof_firstaid_expert ) ? prof_bonus + 2 : prof_bonus;
```
Comment (kevingranade) [inline src/game.cpp:5104]:
```suggestion
        prof_bonus = u.has_proficiency( proficiency_prof_wound_care ) ? prof_bonus + 1 : prof_bonus;
        prof_bonus = u.has_proficiency( proficiency_prof_wound_care_expert ) ? prof_bonus + 2 : prof_bonus;
```

File: src/iuse_actor.cpp [hunk 21]
```diff
@@ -124,6 +124,8 @@ static const itype_id itype_syringe( "syringe" );
 static const mutation_category_id mutation_category_CHIMERA( "CHIMERA" );
 static const mutation_category_id mutation_category_ELFA( "ELFA" );
 
+static const proficiency_id proficiency_prof_firstaid( "prof_firstaid" );
+static const proficiency_id proficiency_prof_firstaid_expert( "prof_firstaid_expert" );
 static const proficiency_id proficiency_prof_traps( "prof_traps" );
 static const proficiency_id proficiency_prof_trapsetting( "prof_trapsetting" );
```
Comment (kevingranade) [inline src/iuse_actor.cpp:130]:
```suggestion
static const proficiency_id proficiency_prof_traps( "prof_traps" );
static const proficiency_id proficiency_prof_trapsetting( "prof_trapsetting" );
static const proficiency_id proficiency_prof_wound_care( "prof_wound_care" );
static const proficiency_id proficiency_prof_wound_care_expert( "prof_wound_care_expert" );
```

File: src/iuse_actor.cpp [hunk 22]
```diff
@@ -3339,17 +3341,14 @@ cata::optional<int> heal_actor::use( Character &p, item &it, bool, const tripoin
...
 
+    // each tier of proficiency cuts requred time by half
     int cost = move_cost;
-    if( long_action ) {
-        // A hack: long action healing on NPCs isn't done yet.
-        // So just heal at start and paralyze the player for 5 minutes.
-        cost /= std::min( 10, p.get_skill_level( skill_firstaid ) + 1 );
-    }
+    cost = p.has_proficiency( proficiency_prof_firstaid_expert ) ? cost / 2 : cost;
+    cost = p.has_proficiency( proficiency_prof_firstaid ) ? cost / 2 : cost;
```
Comment (kevingranade) [inline src/iuse_actor.cpp:3347]:
```suggestion
    cost = p.has_proficiency( proficiency_prof_wound_care_expert ) ? cost / 2 : cost;
    cost = p.has_proficiency( proficiency_prof_wound_care ) ? cost / 2 : cost;
```

File: src/iuse_actor.cpp [hunk 23]
```diff
@@ -3393,8 +3392,12 @@ int heal_actor::get_heal_value( const Character &healer, bodypart_id healed ) co
 int heal_actor::get_bandaged_level( const Character &healer ) const
 {
     if( bandages_power > 0 ) {
+        int prof_bonus = healer.get_skill_level( skill_firstaid );
+        prof_bonus = healer.has_proficiency( proficiency_prof_firstaid ) ? prof_bonus + 1 : prof_bonus;
+        prof_bonus = healer.has_proficiency( proficiency_prof_firstaid_expert ) ? prof_bonus + 2 :
+                     prof_bonus;
```
Comment (kevingranade) [inline src/iuse_actor.cpp:3398]:
```suggestion
        prof_bonus = healer.has_proficiency( proficiency_prof_wound_care ) ? prof_bonus + 1 : prof_bonus;
        prof_bonus = healer.has_proficiency( proficiency_prof_wound_care_expert ) ?
                     prof_bonus + 2 : prof_bonus;
```

File: src/iuse_actor.cpp [hunk 24]
```diff
@@ -3404,7 +3407,11 @@ int heal_actor::get_disinfected_level( const Character &healer ) const
 {
     if( disinfectant_power > 0 ) {
         /** @EFFECT_FIRSTAID increases healing item effects */
-        return disinfectant_power + disinfectant_scaling * healer.get_skill_level( skill_firstaid );
+        int prof_bonus = healer.get_skill_level( skill_firstaid );
+        prof_bonus = healer.has_proficiency( proficiency_prof_firstaid ) ? prof_bonus + 1 : prof_bonus;
+        prof_bonus = healer.has_proficiency( proficiency_prof_firstaid_expert ) ? prof_bonus + 2 :
+                     prof_bonus;
```
Comment (kevingranade) [inline src/iuse_actor.cpp:3413]:
```suggestion
        prof_bonus = healer.has_proficiency( proficiency_prof_wound_care ) ?
                     prof_bonus + 1 : prof_bonus;
        prof_bonus = healer.has_proficiency( proficiency_prof_wound_care_expert ) ?
                     prof_bonus + 2 : prof_bonus;
```

File: src/iuse_actor.cpp [hunk 25]
```diff
@@ -3414,7 +3421,11 @@ int heal_actor::get_stopbleed_level( const Character &healer ) const
 {
     if( bleed > 0 ) {
         /** @EFFECT_FIRSTAID increases healing item effects */
-        return bleed + healer.get_skill_level( skill_firstaid ) / 2;
+        int prof_bonus = healer.get_skill_level( skill_firstaid ) / 2;
+        prof_bonus = healer.has_proficiency( proficiency_prof_firstaid ) ? prof_bonus + 1 : prof_bonus;
+        prof_bonus = healer.has_proficiency( proficiency_prof_firstaid_expert ) ? prof_bonus + 2 :
+                     prof_bonus;
```
Comment (kevingranade) [inline src/iuse_actor.cpp:3427]:
```suggestion
        prof_bonus = healer.has_proficiency( proficiency_prof_wound_care ) ?
                      prof_bonus + 1 : prof_bonus;
        prof_bonus = healer.has_proficiency( proficiency_prof_wound_care_expert ) ?
                     prof_bonus + 2 : prof_bonus;
```

File: src/iuse_actor.cpp [hunk 26]
```diff
@@ -3547,6 +3558,10 @@ int heal_actor::finish_using( Character &healer, Character &patient, item &it,
     practice_amount = std::max( 9.0f, practice_amount );
 
     healer.practice( skill_firstaid, static_cast<int>( practice_amount ) );
+    healer.practice_proficiency( proficiency_prof_firstaid,
+                                 time_duration::from_turns( practice_amount ) );
+    healer.practice_proficiency( proficiency_prof_firstaid_expert,
+                                 time_duration::from_turns( practice_amount ) );
```
Comment (kevingranade) [inline src/iuse_actor.cpp:3564]:
```suggestion
    healer.practice_proficiency( proficiency_prof_wound_care,
                                 time_duration::from_turns( practice_amount ) );
    healer.practice_proficiency( proficiency_prof_wound_care_expert,
                                 time_duration::from_turns( practice_amount ) );
```

File: src/iuse_actor.cpp [hunk 27]
```diff
@@ -3563,9 +3578,11 @@ static bodypart_id pick_part_to_heal(
     const bool precise = &healer == &patient ?
                          patient.has_trait( trait_SELFAWARE ) :
                          /** @EFFECT_PER slightly increases precision when using first aid on someone else */
-
                          /** @EFFECT_FIRSTAID increases precision when using first aid on someone else */
-                         ( healer.get_skill_level( skill_firstaid ) * 4 + healer.per_cur >= 20 );
+                         ( ( healer.get_skill_level( skill_firstaid ) +
+                             ( healer.has_proficiency( proficiency_prof_firstaid ) ? 0 : 1 ) +
+                             ( healer.has_proficiency( proficiency_prof_firstaid ) ? 0 : 2 ) ) * 4 +
+                           healer.per_cur >= 20 );
```
Comment (kevingranade) [inline src/iuse_actor.cpp:3585]:
```suggestion
                         ( ( healer.get_skill_level( skill_firstaid ) +
                             ( healer.has_proficiency( proficiency_prof_wound_care ) ? 0 : 1 ) +
                             ( healer.has_proficiency( proficiency_prof_wound_care ) ? 0 : 2 ) ) * 4 +
                           healer.per_cur >= 20 );
```

File: src/player_hardcoded_effects.cpp [hunk 28]
```diff
@@ -108,6 +108,9 @@ static const mutation_category_id mutation_category_MYCUS( "MYCUS" );
 static const mutation_category_id mutation_category_RAT( "RAT" );
 static const mutation_category_id mutation_category_TROGLOBITE( "TROGLOBITE" );
 
+static const proficiency_id proficiency_prof_firstaid( "prof_firstaid" );
+static const proficiency_id proficiency_prof_firstaid_expert( "prof_firstaid_expert" );
```
Comment (kevingranade) [inline src/player_hardcoded_effects.cpp:112]:
```suggestion
static const proficiency_id proficiency_prof_wound_care( "prof_wound_care" );
static const proficiency_id proficiency_prof_wound_care_expert( "prof_wound_care_expert" );
```

File: src/player_hardcoded_effects.cpp [hunk 29]
```diff
@@ -266,8 +269,12 @@ static void eff_fun_bleed( Character &u, effect &it )
     // QuikClot or bandages per the recipe.)
     const int intense = it.get_intensity();
     // tourniquet reduces effective bleeding by 2/3 but doesn't modify the effect's intensity
+    // proficiency improves that factor to 3/4 and 4/5 respectively
     bool tourniquet = u.worn_with_flag( STATIC( flag_id( "TOURNIQUET" ) ),  it.get_bp() );
-    if( !( tourniquet && one_in( 3 ) ) && u.activity.id() != ACT_FIRSTAID ) {
+    int prof_bonus = 3;
+    prof_bonus = u.has_proficiency( proficiency_prof_firstaid ) ? prof_bonus + 1 : prof_bonus;
+    prof_bonus = u.has_proficiency( proficiency_prof_firstaid_expert ) ? prof_bonus + 1 : prof_bonus;
```
Comment (kevingranade) [inline src/player_hardcoded_effects.cpp:276]:
```suggestion
    prof_bonus = u.has_proficiency( proficiency_prof_wound_care ) ? prof_bonus + 1 : prof_bonus;
    prof_bonus = u.has_proficiency( proficiency_prof_wound_care_expert ) ? prof_bonus + 1 : prof_bonus;
```

File: src/character.cpp [hunk 30]
```diff
@@ -340,6 +340,8 @@ static const proficiency_id proficiency_prof_parkour( "prof_parkour" );
 static const proficiency_id proficiency_prof_spotting( "prof_spotting" );
 static const proficiency_id proficiency_prof_traps( "prof_traps" );
 static const proficiency_id proficiency_prof_trapsetting( "prof_trapsetting" );
+static const proficiency_id proficiency_prof_wound_care( "prof_wound_care" );
+static const proficiency_id proficiency_prof_cound_care_expert( "prof_wound_care_expert" );
```
Comment (kevingranade) [inline src/character.cpp:344]:
```suggestion
static const proficiency_id proficiency_prof_wound_care_expert( "prof_wound_care_expert" );
```

File: src/iuse_actor.cpp [hunk 31]
```diff
@@ -3414,7 +3423,12 @@ int heal_actor::get_stopbleed_level( const Character &healer ) const
 {
     if( bleed > 0 ) {
         /** @EFFECT_FIRSTAID increases healing item effects */
-        return bleed + healer.get_skill_level( skill_firstaid ) / 2;
+        int prof_bonus = healer.get_skill_level( skill_firstaid ) / 2;
+        prof_bonus = healer.has_proficiency( proficiency_prof_wound_care ) ?
+                      prof_bonus + 1 : prof_bonus;
```
Comment (kevingranade) [inline src/iuse_actor.cpp:3428]:
```suggestion
                     prof_bonus + 1 : prof_bonus;
```

**Comment by nexusmrsep** [comment]
Should the skill 'firstaid' be renamed too?

**Comment by kevingranade** [comment]
No, the skill firstaid is in fact a catchall for all manner of first aid related activities such as bandaging and pharmacuticals and bonesetting.

**Pull request merged**